docs/workspaces.md

```diff
@@ -137,38 +150,53 @@ spec:
     mountPath: /custom/path/relative/to/root
 ```
 
-#### Sharing `Workspaces` with `Sidecars`
+#### Isolating `Workspaces` to Specific `Steps` or `Sidecars`
 
-A `Task's` `Sidecars` are also able to access the `Workspaces` the `Task` defines but must have their
-`volumeMount` configuration set explicitly. Below is an example `Task` that shares a `Workspace` between
-its `Steps` and its `Sidecar`. In the example a `Sidecar` sleeps for a short amount of time and then writes
-a `ready` file which the `Step` is waiting for:
+To limit access to a `Workspace` from a subset of a `Task's` `Steps` or `Sidecars` requires
+adding a `workspaces` declaration to those sections. In the following example a `Task` has several
+`Steps` but only the one that performs a `git clone` will be able to access the SSH credentials
+passed into it:
 
 ```yaml
 spec:
   workspaces:
-  - name: signals
+  - name: ssh-credentials
+    description: An .ssh directory with keys, known_host and config files used to clone the repo.
   steps:
-  - image: alpine
-    script: |
-      while [ ! -f "$(workspaces.signals.path)/ready" ]; do
-        echo "Waiting for ready file..."
-        sleep 1
-      done
-      echo "Saw ready file!"
-  sidecars:
-  - image: alpine
-    # Note: must explicitly include volumeMount for the workspace to be accessible in the Sidecar
-    volumeMounts:
-    - name: $(workspaces.signals.volume)
-      mountPath: $(workspaces.signals.path)
-    script: |
-      sleep 3
-      touch "$(workspaces.signals.path)/ready"
+  - name: clone-repo
+    workspaces:
+    - name: ssh-credentials # This Step receives the sensitive workspace; the others do not.
+    image: git
+    script: # git clone ...
+  - name: build-source
+    image: third-party-source-builder:latest # This image doesn't get access to ssh-credentials.
+  - name: lint-source
+    image: third-party-source-linter:latest # This image doesn't get access to ssh-credentials.
 ```
 
-**Note:** `Sidecars` _must_ explicitly opt-in to receiving the `Workspace` volume. Injected `Sidecars` from
-non-Tekton sources will not receive access to `Workspaces`.
+It can potentially be useful to mount `Workspaces` to different locations on a per-`Step` or
+per-`Sidecar` basis and this is also supported:
+
+```yaml
+kind: Task
+spec:
+  workspaces:
+  - name: ws
+    mountPath: /workspaces/ws
+  steps:
+  - name: edit-files-1
+    workspaces:
+    - name: foo
+      mountPath: /foo # overrides mountPath
+  - name: edit-files-2
+    workspaces:
+    - name: foo # no mountPath specified so will use /workspaces/ws
+  sidecars:
+  - name: watch-files-on-workspace
+    workspaces:
+    - name: foo
+      mountPath: /files # overrides mountPath
+```
```

Comment:
the workspace `foo` in each step is referring to the same workspace `ws` specified under `workspaces`? should they be named same as defined under `workspaces`?

Reply:
What does "override" mean here? the step `edit-files-1` is executed with the mount path `/foo` and `edit-files-2` executed with the default mountPath `/ws`. But the files created in `edit-files-1` under `/foo` Will be visible in `edit-files-2` under `/ws`. Its just the path name which differs. Am I getting it right?

Reply:
I ran a pipeline with creating files (`/foo/sample1` and `/foo/sample2`) in `edit-files-1`. Both files were accessible as `/ws/sample1` and `/ws/sample2` in `edit-files-2`.

I don't see any issues with this behavior but may be a one liner explanation might help in the doc. Not a blocker though. 🙏

Reply:
They should be `name`d "ws".  That's a bug, well spotted! will fix.  Also I will make this a validation error if it isn't already.  "foo" isn't a valid workspace name here.